### DISCUSSION TITLE
Can R-universe become a CRAN alternative?

### DISCUSSION THREAD

**Discussion by JosiahParry** [r-universe-org/help#459] [Ideas]
Publication on CRAN is important for a lot of organization and a lot of folks. However, over time there is a repeated pattern of CRAN maintainers arbitrarily removing packages from CRAN, being overtly rude among other things that make it challenging to be a contributor to the R community. 

I believe that the r-universe has the potential to be an R package repository that respects developers and provides autonomy and transparency in the process. However, there are a number of benefits that CRAN provides that I do not think r-universe yet does. 

My goal with this is to start a discussion on what would be needed to create an authoritative R package repository based on the r-universe framework. 

## r-universe benefits:

- Automated and transparent build process
- linux, mac, windows, and webR binaries 
- package metrics and telemetry
- public API
- developer friendly

## What r-universe needs

These are the things that I think that r-universe is missing that CRAN does solve

- archived versions (when a git repo or history is modified the git sha is not sufficient)
- reverse dependency checks 
- documented long term support for R-universe
- message of failing checks in r-devel
- integration tests (R CMD check with presently released versions) that lead to removal if failing without fixing within timely manner 
    - while annoying, i think this is the true benefit that CRAN provides. That every R package that is published at the moment can be installed successfully

**Comment by JosiahParry** [comment]
Related:
- https://github.com/r-universe-org/help/discussions/399
- https://github.com/cranchange/cran_change.org/
- https://github.com/RConsortium/r-repositories-wg

**Comment by jeroen** [comment]
It is not the mission of R-universe to replace CRAN or exactly copy its functionality. Everyone is free to run their own cran-style package repository, with their own policies, for packages that are out of scope or otherwise not suitable for CRAN. This is also emphasised by CRAN founder [Kurt Hornik at UseR 2024](https://jeroen.github.io/images/user2024kurt.jpg) ([all slides](https://static.sched.com/hosted_files/userconf2024/41/user2024_CRAN.pdf)). Organizations like bioconductor and rOpenSci have been doing this for a long time, R-universe provides infrastructure to make this easy for everyone.

You might be interested in the [r-multiverse project](http://r-multiverse.org/) which is a collaborative effort to start a democratic community-managed package repository with quarterly production snapshots, based on r-universe. See this discussion to get up to speed: https://github.com/r-multiverse/help/issues/78
